test/unit/logic/broadcaster.js

```diff
@@ -0,0 +1,242 @@
+			transactions: {
+				transactionInPool: sinonSandbox.stub().returns(false),
+			},
+		};
+
+		broadcaster = new Broadcaster(
+			broadcasts,
+			force,
+			peersStub,
+			transactionStub,
+			loggerStub
+		);
+		broadcaster.bind(
+			modulesStub.peers,
+			modulesStub.transport,
+			modulesStub.transactions
+		);
+		done();
+	});
+
+	after(() => {
+		return sinonSandbox.restore();
+	});
+
+	describe('constructor', () => {
+		it('should return Broadcaster instance', () => {
+			expect(broadcaster).to.be.instanceOf(Broadcaster);
+			expect(broadcaster)
+				.to.have.property('queue')
+				.that.is.an('Array');
+			expect(broadcaster)
+				.to.have.property('config')
+				.that.is.an('object');
+			return expect(broadcaster)
+				.to.have.property('routes')
+				.that.is.an('Array');
+		});
+	});
+
+	describe('enqueue', () => {
+		it('should throw error for no params', () => {
+			return expect(() => {
+				broadcaster.enqueue();
+			})
+				.to.throw()
+				.to.be.instanceOf(Error);
+		});
+
+		it('should push params and options to queue', () => {
+			const params = {};
+			const options = {};
+			expect(broadcaster.enqueue(params, options)).to.eql(1);
+			return expect(broadcaster.enqueue(params, options)).to.eql(2);
+		});
+	});
+
+	describe('getPeers', () => {
+		it('should return peers', done => {
+			broadcaster.getPeers(params, (err, peers) => {
+				expect(err).to.be.null;
+				expect(peers).to.be.an('Array').that.is.not.empty;
+				done();
+			});
+		});
+	});
+
+	describe('broadcast', () => {
+		it('should be able to broadcast block to peers', done => {
+			modulesStub.peers.list.callsArgWith(1, null, peerList);
+			broadcaster.broadcast(params, options, (err, res) => {
+				expect(err).to.be.null;
+				expect(res).to.be.an('object').that.is.not.empty;
+				expect(res).to.deep.equal({ body: null, peer: peerList });
+				done();
+			});
+		});
+
+		it('should throw error for empty peers', done => {
+			const err = new Error('empty peer list');
+			modulesStub.peers.list.callsArgWith(1, err, []);
+			broadcaster.broadcast(params, options, (err, res) => {
+				expect(err).to.be.eql(err);
+				expect(res).to.be.an('object').that.is.not.empty;
+				done();
+			});
+		});
+	});
+
+	describe('maxRelays', () => {
+		it('should return true if exhausted', () => {
+			return expect(broadcaster.maxRelays({ relays: 11 })).to.be.true;
+		});
+
+		it('should return false if max relay is less than relay limit', () => {
+			return expect(broadcaster.maxRelays({ relays: 9 })).to.be.false;
+		});
+	});
+
+	describe('nextRelease', () => {
+		it('should be able to invoke next release', done => {
+			const nextRelease = Broadcaster.__get__('nextRelease');
+			nextRelease(() => {
+				expect(loggerStub.error.called).to.be.true;
+				done();
+			});
+		});
+	});
+
+	describe('filterQueue', () => {
+		it('should be able to filter the queue', done => {
+			const filterQueue = Broadcaster.__get__('__private.filterQueue');
+			filterQueue(() => {
```

Comment:
Would be nice if we have tests for transactions, and blocks and later check if `filterQueue` is working fine or not. This test does not tell us anything about whether the `filterQueue` is a valid function or not.